src/field_map_change.c

```diff
@@ -427,179 +418,175 @@ void sub_02053494 (FieldSystem * fieldSystem)
+static BOOL FieldTask_LoadMapFromContinueGame (TaskManager * taskMan)
 {
     FieldSystem * fieldSystem = TaskManager_FieldSystem(taskMan);
-    UnkStruct_020507E4 * v1 = SaveData_Events(fieldSystem->saveData);
-    int * v2 = sub_02050A68(taskMan);
+    FieldEvents * events = SaveData_GetFieldEvents(fieldSystem->saveData);
+    int * state = FieldTask_GetState(taskMan);
 
-    switch (*v2) {
+    switch (*state) {
     case 0:
         sub_0202878C(fieldSystem->saveData);
 
-        if (sub_0202B6A4(sub_0202B628(fieldSystem->saveData), inline_020535E8(v1))) {
+        if (Journal_CheckOpenOnContinue(SaveData_GetJournal(fieldSystem->saveData), inline_020535E8(events))) {
             sub_0203D30C(fieldSystem, NULL);
-            (*v2) = 4;
+            (*state) = 4;
             break;
         }
     case 1:
-        fieldSystem->unk_9C = sub_0202B634(sub_0202B628(fieldSystem->saveData), inline_020535E8(v1));
+        fieldSystem->unk_9C = Journal_GetSavedPage(SaveData_GetJournal(fieldSystem->saveData), inline_020535E8(events));
 
-        if (sub_0206ADBC(v1)) {
-            UnkStruct_0203A790 * v3 = sub_0203A790(fieldSystem->saveData);
+        if (FieldEvents_CheckCommClub(events)) {
+            FieldStatus * fieldStatus = SaveData_GetFieldStatus(fieldSystem->saveData);
 
-            if (sub_02053518(fieldSystem)) {
-                sub_02053540(fieldSystem);
+            if (FieldSystem_IsSavedInUnionRoom(fieldSystem)) {
+                FieldSystem_SetLocationToUnionRoomExit(fieldSystem);
             }
 
-            sub_0206ADAC(v1);
-            sub_0205311C(fieldSystem, sub_0203A730(v3));
+            FieldEvents_ResetCommClub(events);
+            FieldSystem_MapChange_SetNewLocation(fieldSystem, sub_0203A730(fieldStatus));
             sub_020533CC(fieldSystem);
-            sub_020531C0(fieldSystem, 0);
-            sub_02053320(fieldSystem);
+            FieldSystem_MapChange_UpdateGameData(fieldSystem, 0);
+            FieldSystem_MapChange_CreateObjects(fieldSystem);
         } else {
-            sub_0205311C(fieldSystem, NULL);
+            FieldSystem_MapChange_SetNewLocation(fieldSystem, NULL);
             sub_020533CC(fieldSystem);
             sub_020559DC(fieldSystem);
-            sub_0205338C(fieldSystem);
+            FieldSystem_MapChange_LoadObjects(fieldSystem);
         }
 
         sub_02053494(fieldSystem);
         sub_020705CC(fieldSystem);
-        (*v2) = 2;
+        (*state) = 2;
         break;
     case 2:
-        sub_020559CC(taskMan);
-        (*v2) = 3;
+        FieldTask_StartFadeIn(taskMan);
+        (*state) = 3;
         break;
     case 3:
-        return 1;
+        return TRUE;
     case 4:
         if (!(sub_020509B4(fieldSystem))) {
-            (*v2) = 1;
+            (*state) = 1;
         }
         break;
     }
 
-    return 0;
+    return FALSE;
 }
 
-void sub_02053704 (FieldSystem * fieldSystem)
+void FieldSystem_SetLoadMapFromContinueGameTask (FieldSystem * fieldSystem)
```

Comment:
This name feels a little cumbersome to me, but I think you'd also need to update the name of `FieldTask_Set` to `FieldTask_Start` or something similar, and that runs into a naming conflict.

Reply:
There's already a FieldTask_Start, they differ in params and _Set only runs if there's an already existing taskman (in unk_10) in the FieldSystem struct, and overwrites that taskman in favor of the new one created for the new task. Start will additionally set FieldSystem->unk_00 as the given task manager param

Reply:
Pretty much exactly what I mean, yeah. I think this name is fine for now, even if I'm not *thrilled* about it. I can get a sense of what it does, at least.